src/main/java/net/dv8tion/jda/api/entities/guild/SystemChannelFlag.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.entities.guild;
+
+import javax.annotation.Nonnull;
+import java.util.EnumSet;
+import java.util.HashMap;
+
+/**
+ * Represents Discord's system channel flags of a guild. A system channel is a
+ * unique channel in a guild where guild notices such as welcome messages and boost events
+ * are posted. System channel flags configure suppressions of these messages and events. According
+ * to the <a href="https://discord.com/developers/docs/">Discord API docs<a/>, system channel flags
+ * are represented as one-hot encoded bitmasks, where each bit in a binary string represents whether
+ * a particular flag is enabled. This class contains methods to convert these bitmasks to an {@link EnumSet}
+ * of this class, which is the representation of system channel flags used by users of JDA.
```

Comment:
```suggestion
 * Flags which configures a {@linkplain Guild#getSystemChannel() system channel} of a {@link Guild}.
```

1. The explanation of a system channel is already in `Guild#getSystemChannel()`
2. It is technically correct right now to say system channel flags are for suppressions , but it may not be in the future, so I removed it
3. It is not necessary to explain how the flags are encoded, the user will often encounter them in a `Set<SystemChannelFlag>`
4. The conversion methods are already discoverable in your IDE and the Javadocs
5. Try to split large one liners